### PR TITLE
Use `angular-scroll` for `rb-scrollspy`

### PR DESCRIPTION
**Pull request by chrisvxd**
The scrollspy provided by `angular-strap` did not allow for custom elements for scroll observation, and only allowed for observation on `window/document`, making it unusable in modals or overlays.

- Gutted out angular-strap and replaced with angular-scroll
- Pass in the `id` of the element to observe scrolling via `scroll-id` attr.
- Scroll-spy now animates. Sexy.
- Tests fixed
- Demos fixed

See `angular-scroll` here https://github.com/oblador/angular-scroll.

@ohskylab wasn't sure if I should put the base as `2.0.0` or `master` but feel free to edit it. I would like to see this in the next release!

TP https://rockabox.tpondemand.com/entity/9618

### CODE REVIEW

**Comment by ohskylab** [comment]
@chrisvxd Excellent, fast work! Yes to `2.0.0`, but the base should be `master` so will need to re-do this as a new PR.

File: src/rb-scrollspy/demo/demo.tpl.html [hunk 1]
```diff
@@ -1,12 +1,31 @@
+<style type="text/css">
+  #container {
```
Comment (ohskylab) [inline src/rb-scrollspy/demo/demo.tpl.html]: @chrisvxd I know this is a) just a demo and b) probably pulled from pulled from `angular-scroll` and c) inline, but is it worth making the CSS more SUIT-like?

File: src/rb-scrollspy/rb-scrollspy.tpl.html [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+                <span>{{category.label}}</span>
             </a>
 
             <ul>
-                <li ng-repeat="section in category.items track by $index">
-                    <a class="Scrollspy-Section" href="#{{::section.anchor}}" bs-scrollspy data-target="#{{::section.anchor}}" data-offset="{{::offset || 0}}">
-                        <span>{{::section.label}}</span>
+                <li ng-repeat="section in category.items">
+                    <a class="Scrollspy-Section"
+                        href="#{{section.anchor}}"
```
Comment (k0nG) [inline src/rb-scrollspy/rb-scrollspy.tpl.html]: Think this should be `{{::section.anchor}}`

File: src/rb-scrollspy/demo/demo.tpl.html (hunk 1 above)
Comment (ohskylab) [inline src/rb-scrollspy/demo/demo.tpl.html]: :ok_hand: with this.

**Pull request closed without merging**